Sources/JWTKit/ECDSA/ECDSAKey.swift

```diff
@@ -105,6 +105,28 @@ public struct ECDSAKey<Curve>: ECDSAKeyType where Curve: ECDSACurveType {
         return try self.private(pem: string)
     }
 
+    /// Exports the current public key as a PEM encoded string.
+    ///
+    /// - Throws: If the key is not a public key.
```

Comment:
Your logic, both here and down in `.init(privateKey:publicKey:)` does not account for the fact that a public key can always be derived from a private key (whereas the logic in `generate()` _does_ account for this).